src/Microsoft.ML.LightGBM/WrappedLightGbmTraining.cs

```diff
@@ -33,9 +33,9 @@ internal static class WrappedLightGbmTraining
             double bestScore = double.MaxValue;
             double factorToSmallerBetter = 1.0;
 
-            if (earlyStoppingRound > 0 && (parameters["metric"] == "auc"
-                || parameters["metric"] == "ndcg"
-                || parameters["metric"] == "map"))
+            if (earlyStoppingRound > 0 && (parameters["metric"].ToString() == "auc"
```

Comment:
>parameters["metric"].ToString() [](start = 43, length = 31)

What are these originally?

If they're strings, just use a direct cast. That communicates that you expect them to actually *be* strings. `ToString()` suggests there's some conversion going on.

Reply:
Here it makes to cast to string because a string is represented as a string unlike int, double, etc that can be represented in their original type or as strings.

---
In reply to: [199613266](https://github.com/dotnet/machinelearning/pull/454#discussion_r199613266) [](ancestors = 199613266)